### ISSUE TITLE
`typescript/prefer-optional-chain` difference in behavior with original ESLint rule when performing undefined and null check

### ISSUE THREAD

**Issue by connorshea** [oxc-project/tsgolint#616]
tsgolint v0.11.3

This counts as a violation in tsgolint, but not TS ESlint.

[Playground link for TS ESLint, no error](https://typescript-eslint.io/play/#ts=5.9.2&fileType=.tsx&code=KYDwDg9gTgLgBAMwK4DsDGMCWEWIhACgDcBDAGwGcAuOAbzgFtMUaUkGAjYKOAHzlQATYAmbBBAbjgBfAJR0AUHDiYEcYuQoA6JrgCEAXgMCUw0SnFwAZFbilKO5nEPG2ZMvNpLlcKMBhIULgwUEjAEt7SCt5%2BAUGImuEK0hFAA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Y6RAM0WloHsalfkwCG8WmQAWo5uii9o-aJHBgAviHVA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3QAacDToMmeCmFHopAXxAqgA&tokens=false)

Arbitrary code:

```ts
export function foo(vals: { min: number | undefined; }) {
  if (vals.min !== undefined && vals.min !== null) {
    return true;
  }

  return false;
};
```

ESLint does not care/error on the above code at all. But in tsgolint, we do:

```
× typescript-eslint(prefer-optional-chain): Prefer using an optional chain expression instead, as it's more concise and easier to read.
╭─[foo.tsx:2:7]
1 │ export function foo(vals: { min: number | undefined; }) {
2 │   if (vals.min !== undefined && vals.min !== null) {
·       ───────────────────────────────────────────
3 │     return true;
╰────

Found 0 warnings and 1 error.
```
